tests/components/application-connector/docs/compass-runtime-agent-tests.md

```diff
@@ -0,0 +1,135 @@
+# Compass Runtime Agent
+
+**Table of Contents**
+
+- [Design and architecture](#design-and-architecture)
+- [Building](#building)
+- [Running](#running)
+  - [Deploy a Kyma cluster locally](#deploy-a-kyma-cluster-locally)
+  - [Test setup](#test-setup---compass-runtime-agent-configuration)
+  - [Run the tests](#run-the-tests)
+- [Debugging](#debugging)
+  - [Running without cleanup](#running-without-cleanup)
+
+## Design and architecture
+
+The tests consist of:
+- [Test resources](../resources/charts/compass-runtime-agent-test/) used to perform the test
+- [Test runner](../test/application-connectivity-validator/) with all the test cases
+
+The tests are executed as a Kubernetes Job on a Kyma cluster where the tested Compass Runtime Agent is installed. The test Job is deployed in the `test` Namespace.
+
+![Compass Runtime Agent tests architecture](assets/compass-runtime-agent-tests-architecture.svg)
+
+The interactions between components are the following:
+
+1. Compass Runtime Agent periodically fetches certificates from Compass Connector.
+2. Compass Runtime Agent periodically fetches applications from Compass Director.
+3. Compass Runtime Agent Test sends GraphQL mutations to Compass Director to create, modify, or delete Applications.
+4. Compass Runtime Agent Test verifies whether corresponding Application CRs were created, modified, or deleted.
+5. Compass Runtime Agent Test verifies whether secret with certificates used for communication with Director was created.
+6. Compass Runtime Agent Test verifies whether secret with CA root certificate used by Istio Gateway was created.
+7. Compass Runtime Agent Test verifies CompassConnection CR contents.
+
+## Building
+
+Pipelines build the Compass Runtime Agent test using the **release** target from the `Makefile`.
+
+To build **and push** the Docker images of the tests, run:
+
+``` sh
+./scripts/local-build.sh {DOCKER_TAG} {DOCKER_PUSH_REPOSITORY}
+```
+
+This will build the following images:
+- `{DOCKER_PUSH_REPOSITORY}/compass-runtime-agent-test:{DOCKER_TAG}`
+
+## Running
+
+Tests can be run on any Kyma cluster with Compass Runtime Agent.
+
+Pipelines run the tests using the **test-compass-runtime-agent** target from the `Makefile`.
+
+### Deploy a Kyma cluster locally
+
+1. Provision a local Kubernetes cluster with k3d:
+   ```sh
+   kyma provision k3d
+   ```
```

Comment:
``````suggestion
   ```bash
   kyma provision k3d
   ```
``````

We use `bash` everywhere else for Terminal commands (at least if I review it :D but also in other places in this document), let's be consistent.